_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp

```diff
@@ -4398,18 +4398,21 @@ mfxStatus MfxHwH264Encode::CheckVideoParamQueryLike(
         changed = true;
     }
 
+    bool mfxRateControlCbrOnBxt = 
+        (platform == MFX_HW_APL || platform == MFX_HW_CFL) && 
+        (par.mfx.RateControlMethod == MFX_RATECONTROL_CBR);
+
+    bool mfxRateControl =
+            par.mfx.RateControlMethod == MFX_RATECONTROL_LA  ||
+            par.mfx.RateControlMethod == MFX_RATECONTROL_LA_HRD ||
+            par.mfx.RateControlMethod == MFX_RATECONTROL_LA_EXT ||
+            par.mfx.RateControlMethod == MFX_RATECONTROL_VBR ||
+            par.mfx.RateControlMethod == MFX_RATECONTROL_QVBR ||
+            (mfxRateControlCbrOnBxt && !IsOn(extOpt2->ExtBRC));
```

Comment:
@eistarov : what is this BXT specific check about? What exactly they tried to change for BXT? Is my understanding correct that they enabled sliding window for CBR mode on BXT? But what the check for ExtBrc is about?

Reply:
It looks like they're trying to say: "Sliding window + CBR is supported on BXT with HW BRC, but still not with SW one (aka ExtBRC)".